### PR TITLE
black-macchiato: init at 1.3.0

### PR DESCRIPTION
**Pull request by jperras**
###### Motivation for this change

Adds the Python package `black-macchiato`, which allows the usage of Python's `black` formatter for subsets of an entire file.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4098,6 +4098,12 @@
     githubId = 1918771;
     name = "Joe Doyle";
   };
+  jperras = {
+    email = "joel@nerderati.com";
+    github = "jperras";
+    githubId = 20675;
+    name = "Joël Perras";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:4106]: Please make this into a separate commit:

```
maintainers: add <name>
```

File: pkgs/development/python-modules/black-macchiato/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner  = "wbolster";
+    repo   = pname;
+    rev    = "${version}";
+    sha256 = "0lc9w50nlbmlzj44krk7kxcia202fhybbnwfh77xixlc7vb4rayl";
+  };
+
+  checkPhase = "pytest";
+
+  checkInputs = [ pytest black ];
```
Comment (jonringer) [inline pkgs/development/python-modules/black-macchiato/default.nix:24]:
```suggestion
  checkInputs = [ pytestCheckHook black ];
```

File: pkgs/development/python-modules/black-macchiato/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+buildPythonPackage rec {
+  pname = "black-macchiato";
+  version = "1.3.0";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner  = "wbolster";
+    repo   = pname;
+    rev    = "${version}";
```
Comment (jonringer) [inline pkgs/development/python-modules/black-macchiato/default.nix:18]:
```suggestion
    rev    = "version";
```

File: pkgs/development/python-modules/black-macchiato/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{ stdenv,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonOlder,
+  pytest,
```
Comment (jonringer) [inline pkgs/development/python-modules/black-macchiato/default.nix:5]:
```suggestion
  pytestCheckHook,
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

**Comment by jonringer** [comment]
@jperras thanks for opening your first PR against nixpkgs :)

File: pkgs/development/python-modules/black-macchiato/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+buildPythonPackage rec {
+  pname = "black-macchiato";
+  version = "1.3.0";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner  = "wbolster";
+    repo   = pname;
+    rev    = "version";
```
Comment (jonringer) [inline pkgs/development/python-modules/black-macchiato/default.nix:18]:
```suggestion
    rev    = version;
```

Comment (jonringer) [reply pkgs/development/python-modules/black-macchiato/default.nix:18]: since version is a string to begin with, we don't have to interpolate it, we can just pass the version variable to rev

Comment (jperras) [reply pkgs/development/python-modules/black-macchiato/default.nix:18]: Ah, my mistake – I should have caught that! Fixing.

File: pkgs/development/python-modules/black-macchiato/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner  = "wbolster";
+    repo   = pname;
+    rev    = version;
+    sha256 = "0lc9w50nlbmlzj44krk7kxcia202fhybbnwfh77xixlc7vb4rayl";
+  };
+
+  checkInputs = [ pytestCheckHook black ];
```
Comment (jonringer) [inline pkgs/development/python-modules/black-macchiato/default.nix:22]:
```suggestion
  propagatedBuildInputs = [ black ];

  checkInputs = [ pytestCheckHook ];
```
```
[nix-shell:/home/jon/.cache/nixpkgs-review/pr-96976-1]$ nix-shell --pure -p "with import ./nixpkgs { }; black-macchiato"

[nix-shell:/home/jon/.cache/nixpkgs-review/pr-96976-1]$ black-macchiato --help
Traceback (most recent call last):
  File "/nix/store/33fap5zfxiawb8kmpyy6harfidkdqmig-python3.8-black-macchiato-1.3.0/bin/.black-macchiato-wrapped", line 6, in <module>
    from macchiato import main
  File "/nix/store/33fap5zfxiawb8kmpyy6harfidkdqmig-python3.8-black-macchiato-1.3.0/lib/python3.8/site-packages/macchiato.py", line 8, in <module>
    import black
ModuleNotFoundError: No module named 'black'
```

Comment (jonringer) [reply pkgs/development/python-modules/black-macchiato/default.nix:22]: a `pythonImportsCheck` would have also caught this, do you mind adding one?

Please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

Comment (jperras) [reply pkgs/development/python-modules/black-macchiato/default.nix:22]: Interesting - I previously did have a `propagatedBuildInputs = [ black ];`, and subsequently verified that the binary worked by running `echo "    if True: print('hi')" | ./result/bin/black-macchiato`, which yielded the expected result.

That line must have been mistakenly removed during one of my changes and I then didn't run the resulting binary. My apologies. Fixing.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 96976 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>black-macchiato (python38Packages.black-macchiato)</li>    <li>python37Packages.black-macchiato</li>  </ul></details>
